core/api/v1/keytransparency.proto

```diff
@@ -187,6 +195,35 @@ message ListEntryHistoryResponse {
   int64 next_start = 2;
 }
 
+// BatchListUserRevisionsRequest contains a list of users and a range of revisions.
+message BatchListUserRevisionsRequest {
+  // directory_id identifies the directory in which the user lives.
+  string directory_id = 1;
+  // user_ids are the user identifiers.
+  repeated string user_ids = 2;
+  // start is the starting revision.
+  int64 start = 3;
+  // page_size is the maximum number of entries to return.
+  int32 page_size = 4;
+  // last_verified_tree_size is the tree_size of the last log root the client
+  // verified. Omitting this field will omit the log consistency proof from the
+  // response.
+  int64 last_verified_tree_size = 5;
+}
+
+// BatchListUserRevisionsResponse
+message BatchListUserRevisionsResponse {
+  // latest_log_root contains the lastest log root and its consistency proof.
+  LogRoot latest_log_root = 1;
+  // map_revisions represents the list of keys this user_id has contained over
+  // time. At most page_size results will be returned.
+  repeated BatchMapRevision map_revisions = 2;
+  // next_page_token is a pagination token which will be set if more than
+  // page_size results are available. Clients can pass this value as
```

Comment:
Should the count of results per page be measured in revisions, or revisions * users?

Reply:
That's a great question... I suppose we can answer that a couple of way:

a) In terms of API round trips to Trillian, we have 1 API call per revision because the Trillian API exposes a batch get map leaves at revision API. 
b) In terms of bandwidth and DB access the costs per API call are more similar to revisions * users. 

Since this is a *list* of revisions, I think page size should be measured in revisions. 
Costs are typically more of a quota kind of thing, which I think we should charge in terms of revisions * users. Furthermore, when we implement more efficient ways of querying ranges of revisions we can reduce the quota charge without messing with the API semantics.